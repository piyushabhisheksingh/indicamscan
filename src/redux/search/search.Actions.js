import * as types from './search.Types';
import { stocksprefix } from '../../utils/commonFunctions'

/**
 * searchAction, it takes 3 params and calls the search api function
 */
export const search = (index, input, prevIndex, delay) => {
    /**
    * @param {number|string|number} index input prevIndex - index is the current selectedTab, input is the user search query and prevIndex of selected tab
    */
    let params = stocksprefix(index, input);
    return {
        type: types.SEARCH,
        params,
        scrip: input,
        mode: index,
        prevIndex,
        delay
    }
}

/**
 * clearSearchState, this will not take any argument and it clears the search text input
 */
export const clearSearchState = () => {
    return {
        type: types.CLEAR_SEARCH_STATE
    }
}

/**
 * recentSearch, this will not take any argument and it will set a state for recent search
 */
// export const recentSearch = () => {
//     return {
//         type: types.RECENT_SEARCH
//     }
// }

/**
 * setSearchText, this will take one argument and sets the user search input
 */
export const setSearchText = (payload, key) => {
    /**
    * @param {string} payload - payload is the user search input
    */
    return {
        type: types.SET_SEARCH_TEXT,
        payload,
        key
    }
}

/**
 * setTab, this will take one argument and sets the selected tab
 */
export const setTab = (payload) => {
    /**
   * @param {number} payload - payload is the user selectedTabIndex
   */
    return {
        type: types.SET_SELECTED_TAB,
        payload
    }
}

/**
 * clearState, this will not take any arguments and it clear the result array
 */
export const clearState = () => {
    return {
        type: types.CLEAR_RESULT
    }
}

export const addStock = (payload, onSuccess, onError) => {
    return {
        type: types.ADD_STOCK,
        payload,
        onSuccess,
        onError
    }
}

export const deleteStock = (payload, onSuccess, onError) => {
    return {
        type: types.DELETE_STOCK,
        payload,
        onSuccess,
        onError
    }
}

export const addStockToRecent = (payload) => {
    return {
        type: types.ADD_RECENTSTOCK,
        payload
    }
}

