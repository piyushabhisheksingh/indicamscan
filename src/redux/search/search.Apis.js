import { api } from '../config/api';
import { apiRoutes, BASE_API_URL } from '../config/routes';
import { base64 } from '../../utils/commonFunctions';
import { setLoading, removeLoading } from '../../components/common/LoadingIndicator'
import { services } from '../config/api'
/**
 * search, this will take one argument and it converts params to base64 and then it will call the search api
 */
export const search = (params) => {
    setLoading();
    /**
    * @param {string} params - params contains display segment and search query
    */
    let encoded = base64(params);
    return api(BASE_API_URL).get(apiRoutes.search + encoded)
        .then((response) => {
            removeLoading();
            if (response && response.data) {
                return { response };
            } else {
                return { response: null };
            }
        })
        .catch((error) => {
            removeLoading();
            return { response: error };
        });
};

export const addStock = (params) => {
    setLoading();
    services.apitype = 'addstock';
    return api(BASE_API_URL).post(apiRoutes.addStock(params.watchlistId, params.wtoken))
        .then((response) => {
            removeLoading();
            return response
        })
        .catch((error) => {
            removeLoading();
            return error
        })
}

export const addStockByWname = (params) => {
    setLoading();
    services.apitype = 'addStockByWname';
    return api(BASE_API_URL).post(apiRoutes.addStockByWname(params.watchlistName, params.wtoken))
        .then((response) => {
            removeLoading();
            return response
        })
        .catch((error) => {
            removeLoading();
            return error
        })
}

export const addRecentStock = (params) => {
    services.apitype = 'addstock';
    setLoading();
    return api(BASE_API_URL).post(apiRoutes.recentAddStock(params.watchlistName, params.wtoken))
        .then((response) => {
            removeLoading();
            return response
        })
        .catch((error) => {
            removeLoading();
            return error
        })
}

export const deleteStock = (params) => {
    services.apitype = 'deleteStock';
    setLoading();
    return api(BASE_API_URL).delete(apiRoutes.deleteStock(params.watchlistId, params.wtoken))
        .then((response) => {
            removeLoading();
            return response
        })
        .catch((error) => {
            removeLoading();
            return error
        })
}

/**
 * mock, this will take no arguments and it will call the news api
 */
/*export const mock = () => {
    const url = apiRoutes.mocks[getRandom()].news;
    return mockApi.get(url)
        .then((response) => {
            if (response && response.data) {
                return response;
            } else {
                return null;
            }
        })
        .catch((error) => {
            return error;
        });
}*/