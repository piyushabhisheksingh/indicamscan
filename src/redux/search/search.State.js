export default {
  search: {
    all: [],
    cash: [],
    currency: [],
    commodity: [],
    fno: [],
    selectedTab: 0,
    searchText: '',
    loading: { key: 0, state: false },
    news: [],
    index: 3,
    recentSearchVisible: true,
  }
};