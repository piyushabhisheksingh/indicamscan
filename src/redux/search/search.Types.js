/**
 * action type for search
 */
export const SEARCH = 'SEARCH';
/**
 * action type for selected tab
 */
export const SET_SELECTED_TAB = 'SET_SELECTED_TAB';
/**
 * action type for search text
 */
export const SET_SEARCH_TEXT = 'SET_SEARCH_TEXT'
/**
 * action type for success search api response
 */
export const SEARCH_SUCCESS = 'SEARCH_SUCCESS'
/**
 * action type for failure search api response
 */
export const SEARCH_FAILURE = 'SEARCH_FAILURE'
/**
 * action type for clearing search state
 */
export const CLEAR_SEARCH_STATE = 'CLEAR_SEARCH_STATE'
/**
 * action type for reatining data
 */
export const RETAIN_DATA = 'RETAIN_DATA'
/**
 * action type for clearing search result
 */
export const CLEAR_RESULT = 'CLEAR_RESULT'
/**
 * action type for success news api response
 */
export const NEWS_SUCCESS = 'NEWS_SUCCESS'

/**
 * action type for recent search 
 */
export const RECENT_SEARCH = "RECENT_SEARCH"


export const SET_ENABLE_LOG = 'SET_ENABLE_LOG'

export const ADD_STOCK = 'ADD_STOCK';

export const DELETE_STOCK = 'DELETE_STOCK';

/**
 * action type for addToStock api success response
 */
export const ADDSTOCK_SUCCESS = "ADDSTOCK_SUCCESS"
/**
 * action type for addToStock api failure response
 */
export const ADDSTOCK_FAILURE = "ADDSTOCK_FAILURE"

/**
* Action type for adding stock to recent list
*/
export const ADD_RECENTSTOCK = "ADD_RECENTSTOCK"

