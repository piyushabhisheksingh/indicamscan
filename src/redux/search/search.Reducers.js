import * as types from './search.Types';
import appState from '././search.State';
import lables from '../../utils/lables';
import { DeviceEventEmitter } from 'react-native';
import constants from '../../utils/appConstants'
import { eventName, LogEvents } from '../../../__analytics__/AppAnalytics';

/**
 * searchReducer, The reducer is a pure function that takes the previous state and an action, and returns the next state. (previousState, action) => nextState.
 */
const searchReducer = (state = appState.search, action) => {
    /**
    * @param {Object|string} state, action - state is the previousState and action is the particular action type.
    */
    switch (action.type) {
        case types.SEARCH: {

            if (action.prevIndex != action.mode) {
                if (action.prevIndex == 0) {
                    return {
                        ...state,
                        all: [],
                        news: [],
                        index: 3,
                        loading: { key: 0, state: true },
                        selectedTab: action.mode,
                        searchText: action.scrip,
                        recentSearchVisible: false

                    }
                }
                else if (action.prevIndex == 1) {
                    return {
                        ...state,
                        cash: [],
                        news: [],
                        index: 3,
                        loading: { key: 1, state: true },
                        selectedTab: action.mode,
                        searchText: action.scrip,
                        recentSearchVisible: false

                    }
                }
                else if (action.prevIndex == 2) {
                    return {
                        ...state,
                        fno: [],
                        news: [],
                        index: 3,
                        loading: { key: 2, state: true },
                        selectedTab: action.mode,
                        searchText: action.scrip,
                        recentSearchVisible: false

                    }
                }
                else if (action.prevIndex == 3) {
                    return {
                        ...state,
                        currency: [],
                        news: [],
                        index: 3,
                        loading: { key: 3, state: true },
                        selectedTab: action.mode,
                        searchText: action.scrip,
                        recentSearchVisible: false

                    }
                }
                else {
                    return {
                        ...state,
                        commodity: [],
                        news: [],
                        index: 3,
                        loading: { key: 4, state: true },
                        selectedTab: action.mode,
                        searchText: action.scrip,
                        recentSearchVisible: false

                    }
                }
            } else {
                return {
                    ...state,
                    all: [],
                    cash: [],
                    fno: [],
                    currency: [],
                    commodity: [],
                    news: [],
                    index: 3,
                    loading: { key: action.mode, state: true },
                    selectedTab: action.mode,
                    searchText: action.scrip,
                    recentSearchVisible: false

                }
            }
        }
        case types.SET_SELECTED_TAB: {

            if (state.searchText.length <= 2 && state.searchText.length > 0) {
                return {
                    ...state,
                    selectedTab: action.payload,
                    all: lables.minchar,
                    cash: lables.minchar,
                    currency: lables.minchar,
                    commodity: lables.minchar,
                    fno: lables.minchar,
                    recentSearchVisible: false

                }
            }
            else {
                return {
                    ...state,
                    selectedTab: action.payload,
                    all: [],
                    cash: [],
                    currency: [],
                    commodity: [],
                    fno: [],
                    recentSearchVisible: false

                }
            }
        }

        case types.RECENT_SEARCH: {
            // if (!state.recentSearchVisible){
            DeviceEventEmitter.emit(constants.RECENTSEARCH);
            // }
            return {
                ...state,
                recentSearchVisible: true
            }
        }

        case types.SET_SEARCH_TEXT:
            // if (state.recentSearchVisible){
            DeviceEventEmitter.emit(constants.SEARCH);
            // }
            return {
                ...state,
                searchText: action.payload,
                all: action.payload.length <= 1 ? lables.minchar : [],
                cash: action.payload.length <= 1 ? lables.minchar : [],
                currency: action.payload.length <= 1 ? lables.minchar : [],
                commodity: action.payload.length <= 1 ? lables.minchar : [],
                fno: action.payload.length <= 1 ? lables.minchar : [],
                loading: { key: action.key, state: false },
                recentSearchVisible: false
            }
        case types.SEARCH_SUCCESS: {
            let loading = state.loading;
            loading.state = false
            if (action.mode == 0) {
                return {
                    ...state,
                    all: action.data.result,
                    loading,
                    recentSearchVisible: false

                }
            }
            else if (action.mode == 1) {
                return {
                    ...state,
                    cash: action.data.result,
                    loading,
                    recentSearchVisible: false

                }
            }
            else if (action.mode == 2) {
                return {
                    ...state,
                    fno: action.data.result,
                    loading,
                    recentSearchVisible: false

                }
            }
            else if (action.mode == 3) {
                return {
                    ...state,
                    currency: action.data.result,
                    loading,
                    recentSearchVisible: false

                }
            }
            else {
                return {
                    ...state,
                    commodity: action.data.result,
                    loading,
                    recentSearchVisible: false

                }
            }
        }
        case types.SEARCH_FAILURE: {

            /**
             * Capturing Log Events for Search failure scenario
             */
            LogEvents(eventName.noSearchResult, {search_input: state.searchText})

            let loading = state.loading;
            loading.key = action.mode;
            loading.state = false
            if (action.mode == 0) {
                return {
                    ...state,
                    all: action.data,
                    loading,
                    recentSearchVisible: false

                }
            }
            else if (action.mode == 1) {
                return {
                    ...state,
                    cash: action.data,
                    loading,
                    recentSearchVisible: false

                }
            }
            else if (action.mode == 2) {
                return {
                    ...state,
                    fno: action.data,
                    loading,
                    recentSearchVisible: false

                }
            }
            else if (action.mode == 3) {
                return {
                    ...state,
                    currency: action.data,
                    loading,
                    recentSearchVisible: false

                }
            }
            else {
                return {
                    ...state,
                    commodity: action.data,
                    loading,
                    recentSearchVisible: false

                }
            }
        }
        case types.CLEAR_RESULT: {
            return {
                ...state,
                all: [],
                cash: [],
                currency: [],
                commodity: [],
                fno: [],
                news: [],
                index: 3,
                recentSearchVisible: true

            }
        }
        case types.CLEAR_SEARCH_STATE:
            return {

                all: [],
                cash: [],
                currency: [],
                commodity: [],
                fno: [],
                selectedTab: 0,
                searchText: '',
                loading: { key: 0, state: false },
                news: [],
                index: 3,
                recentSearchVisible: true


            }
        case types.NEWS_SUCCESS:
            return {
                ...state,
                news: action.data,
                index: action.index
            }
        case types.SET_ENABLE_LOG:
            return {
                ...state,
                enableLog: action.payload
            }
        default:
            return state
    }

}
export default searchReducer;