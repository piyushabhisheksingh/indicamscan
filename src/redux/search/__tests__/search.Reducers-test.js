import search from '../search.Reducers'
import * as types from '../search.Types'

const state = {
    all: [],
    cash: [],
    currency: [],
    commodity: [],
    fno: [],
    selectedTab: 0,
    searchText: '',
    loading: false,
    news: [],
    index: 3,
    recentSearchVisible: true,
    event: {},
    enableLog: false
}

const state2 = {
    all: [],
    cash: [],
    currency: [],
    commodity: [],
    fno: [],
    selectedTab: 0,
    searchText: 'a',
    loading: false,
    news: [],
    index: 3,
    recentSearchVisible: true,
    event: {},
    enableLog: false
}
const state3 = {
    all: [],
    cash: [],
    currency: [],
    commodity: [],
    fno: [],
    selectedTab: 0,
    searchText: 'abs',
    loading: false,
    news: [],
    index: 3,
    recentSearchVisible: true,
    event: {},
    enableLog: false
}
const action1 = { type: types.SEARCH, prevIndex: 0, mode: 0, scrip: 'as' }
const action2 = { type: types.SEARCH, prevIndex: 1, mode: 0, scrip: 'as' }
const action3 = { type: types.SEARCH, prevIndex: 2, mode: 0, scrip: 'as' }
const action4 = { type: types.SEARCH, prevIndex: 3, mode: 0, scrip: 'as' }
const action5 = { type: types.SEARCH, prevIndex: 4, mode: 0, scrip: 'as' }
const action6 = { type: types.SET_SELECTED_TAB, payload: 2 }
const action7 = { type: types.SET_SEARCH_TEXT, payload: 'aghs' }
const action8 = { type: types.SEARCH_SUCCESS, data: { result: [] }, mode: 0 }
const action9 = { type: types.SEARCH_SUCCESS, data: { result: [[1, 2]] }, mode: 1 }
const action10 = { type: types.SEARCH_SUCCESS, data: { result: [] }, mode: 2 }
const action11 = { type: types.SEARCH_SUCCESS, data: { result: [] }, mode: 3 }
const action12 = { type: types.SEARCH_SUCCESS, data: { result: [] }, mode: 4 }
const action13 = { type: types.SEARCH_FAILURE, data: { result: [] }, mode: 0 }
const action14 = { type: types.SEARCH_FAILURE, data: { result: [[1, 2]] }, mode: 1 }
const action15 = { type: types.SEARCH_FAILURE, data: { result: [] }, mode: 2 }
const action16 = { type: types.SEARCH_FAILURE, data: { result: [] }, mode: 3 }
const action17 = { type: types.SEARCH_FAILURE, data: { result: [] }, mode: 4 }
const action18 = { type: types.CLEAR_SEARCH_STATE }
const action19 = { type: types.NEWS_SUCCESS, data: [], index: 1 }
const action20 = { type: "ABSB" }
const action21 = { type: types.CLEAR_RESULT }
const action22 = { type: types.SET_SEARCH_TEXT, payload: '' }
const action23 = { type: types.SEARCH, prevIndex: 0, mode: 1, scrip: 'as' }
const action24 = { type: types.SET_ENABLE_LOG, payload: []}
const action25 = { type: types.RECENT_SEARCH }


describe('Search Actions', () => {
    it('1. verify SEARCH', () => {
        const result = search(state, action1)
        expect(result).toMatchSnapshot();
    });
    it('2. verify SEARCH', () => {
        const result = search(state, action2)
        expect(result).toMatchSnapshot();
    });
    it('3. verify SEARCH', () => {
        const result = search(state, action3)
        expect(result).toMatchSnapshot();
    });
    it('4. verify SEARCH', () => {
        const result = search(state, action4)
        expect(result).toMatchSnapshot();
    });
    it('5. verify SEARCH', () => {
        const result = search(state, action5)
        expect(result).toMatchSnapshot();
    });
    it('6. verify SET_SELECTED_TAB', () => {
        const result = search(state, action6)
        expect(result).toMatchSnapshot();
    });
    it('7. verify SET_SELECTED_TAB', () => {
        const result = search(state2, action6)
        expect(result).toMatchSnapshot();
    });
    it('8. verify SET_SELECTED_TAB', () => {
        const result = search(state3, action6)
        expect(result).toMatchSnapshot();
    });
    it('9. verify SET_SEARCH_TEXT', () => {
        const result = search(state3, action7)
        expect(result).toMatchSnapshot();
    });
    it('10. verify SEARCH_SUCCESS when mode is 0', () => {
        const result = search(state3, action8)
        expect(result).toMatchSnapshot();
    });
    it('11. verify SEARCH_SUCCESS when mode is 1', () => {
        const result = search(state3, action9)
        expect(result).toMatchSnapshot();
    });
    it('12. verify SEARCH_SUCCESS when mode is 2', () => {
        const result = search(state3, action10)
        expect(result).toMatchSnapshot();
    });
    it('13. verify SEARCH_SUCCESS when mode is 3', () => {
        const result = search(state3, action11)
        expect(result).toMatchSnapshot();
    });
    it('14. verify SEARCH_SUCCESS when mode is 4', () => {
        const result = search(state3, action12)
        expect(result).toMatchSnapshot();
    });
    it('15. verify SEARCH_FAILURE when mode is 0', () => {
        const result = search(state3, action13)
        expect(result).toMatchSnapshot();
    });
    it('16. verify SEARCH_FAILURE when mode is 1', () => {
        const result = search(state3, action14)
        expect(result).toMatchSnapshot();
    });
    it('17. verify SEARCH_FAILURE when mode is 2', () => {
        const result = search(state3, action15)
        expect(result).toMatchSnapshot();
    });
    it('18. verify SEARCH_FAILURE when mode is 3', () => {
        const result = search(state3, action16)
        expect(result).toMatchSnapshot();
    });
    it('19. verify SEARCH_FAILURE when mode is 4', () => {
        const result = search(state3, action17)
        expect(result).toMatchSnapshot();
    });
    it('20. verify CLEAR_SEARCH_STATE', () => {
        const result = search(state3, action18)
        expect(result).toMatchSnapshot();
    });
    it('21. verify NEWS_SUCCESS', () => {
        const result = search(state3, action19)
        expect(result).toMatchSnapshot();
    });
    it('22. verify default case', () => {
        const result = search(state3, action20)
        expect(result).toMatchSnapshot();
    });
    it('23. verify CLEAR_RESULT', () => {
        const result = search(state3, action21)
        expect(result).toMatchSnapshot();
    });
    it('24. verify SET_SEARCH_TEXT', () => {
        const result = search(state3, action22)
        expect(result).toMatchSnapshot();
    });
    it('25. verify search', () => {
        const result = search(state, action23)
        expect(result).toMatchSnapshot();
    });
    it('26. verify search', () => {
        const result = search(state, action24)
        expect(result).toMatchSnapshot();
    });
     it('26. verify search', () => {
        const result = search(state, action25)
        expect(result).toMatchSnapshot();
    });
});