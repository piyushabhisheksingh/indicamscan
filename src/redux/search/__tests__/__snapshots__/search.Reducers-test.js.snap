// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Actions 1. verify SEARCH 1`] = `
Object {
  "all": Array [],
  "cash": Array [],
  "commodity": Array [],
  "currency": Array [],
  "enableLog": false,
  "event": Object {},
  "fno": Array [],
  "index": 3,
  "loading": Object {
    "key": 0,
    "state": true,
  },
  "news": Array [],
  "recentSearchVisible": false,
  "searchText": "as",
  "selectedTab": 0,
}
`;

exports[`Search Actions 2. verify SEARCH 1`] = `
Object {
  "all": Array [],
  "cash": Array [],
  "commodity": Array [],
  "currency": Array [],
  "enableLog": false,
  "event": Object {},
  "fno": Array [],
  "index": 3,
  "loading": Object {
    "key": 1,
    "state": true,
  },
  "news": Array [],
  "recentSearchVisible": false,
  "searchText": "as",
  "selectedTab": 0,
}
`;

exports[`Search Actions 3. verify SEARCH 1`] = `
Object {
  "all": Array [],
  "cash": Array [],
  "commodity": Array [],
  "currency": Array [],
  "enableLog": false,
  "event": Object {},
  "fno": Array [],
  "index": 3,
  "loading": Object {
    "key": 2,
    "state": true,
  },
  "news": Array [],
  "recentSearchVisible": false,
  "searchText": "as",
  "selectedTab": 0,
}
`;

exports[`Search Actions 4. verify SEARCH 1`] = `
Object {
  "all": Array [],
  "cash": Array [],
  "commodity": Array [],
  "currency": Array [],
  "enableLog": false,
  "event": Object {},
  "fno": Array [],
  "index": 3,
  "loading": Object {
    "key": 3,
    "state": true,
  },
  "news": Array [],
  "recentSearchVisible": false,
  "searchText": "as",
  "selectedTab": 0,
}
`;

exports[`Search Actions 5. verify SEARCH 1`] = `
Object {
  "all": Array [],
  "cash": Array [],
  "commodity": Array [],
  "currency": Array [],
  "enableLog": false,
  "event": Object {},
  "fno": Array [],
  "index": 3,
  "loading": Object {
    "key": 4,
    "state": true,
  },
  "news": Array [],
  "recentSearchVisible": false,
  "searchText": "as",
  "selectedTab": 0,
}
`;

exports[`Search Actions 6. verify SET_SELECTED_TAB 1`] = `
Object {
  "all": Array [],
  "cash": Array [],
  "commodity": Array [],
  "currency": Array [],
  "enableLog": false,
  "event": Object {},
  "fno": Array [],
  "index": 3,
  "loading": false,
  "news": Array [],
  "recentSearchVisible": false,
  "searchText": "",
  "selectedTab": 2,
}
`;

exports[`Search Actions 7. verify SET_SELECTED_TAB 1`] = `
Object {
  "all": "Enter min 2 characters",
  "cash": "Enter min 2 characters",
  "commodity": "Enter min 2 characters",
  "currency": "Enter min 2 characters",
  "enableLog": false,
  "event": Object {},
  "fno": "Enter min 2 characters",
  "index": 3,
  "loading": false,
  "news": Array [],
  "recentSearchVisible": false,
  "searchText": "a",
  "selectedTab": 2,
}
`;

exports[`Search Actions 8. verify SET_SELECTED_TAB 1`] = `
Object {
  "all": Array [],
  "cash": Array [],
  "commodity": Array [],
  "currency": Array [],
  "enableLog": false,
  "event": Object {},
  "fno": Array [],
  "index": 3,
  "loading": false,
  "news": Array [],
  "recentSearchVisible": false,
  "searchText": "abs",
  "selectedTab": 2,
}
`;

exports[`Search Actions 9. verify SET_SEARCH_TEXT 1`] = `
Object {
  "all": Array [],
  "cash": Array [],
  "commodity": Array [],
  "currency": Array [],
  "enableLog": false,
  "event": Object {},
  "fno": Array [],
  "index": 3,
  "loading": Object {
    "key": undefined,
    "state": false,
  },
  "news": Array [],
  "recentSearchVisible": false,
  "searchText": "aghs",
  "selectedTab": 0,
}
`;

exports[`Search Actions 10. verify SEARCH_SUCCESS when mode is 0 1`] = `
Object {
  "all": Array [],
  "cash": Array [],
  "commodity": Array [],
  "currency": Array [],
  "enableLog": false,
  "event": undefined,
  "fno": Array [],
  "index": 3,
  "loading": false,
  "news": Array [],
  "recentSearchVisible": false,
  "searchText": "abs",
  "selectedTab": 0,
}
`;

exports[`Search Actions 11. verify SEARCH_SUCCESS when mode is 1 1`] = `
Object {
  "all": Array [],
  "cash": Array [
    Array [
      1,
      2,
    ],
  ],
  "commodity": Array [],
  "currency": Array [],
  "enableLog": false,
  "event": undefined,
  "fno": Array [],
  "index": 3,
  "loading": false,
  "news": Array [],
  "recentSearchVisible": false,
  "searchText": "abs",
  "selectedTab": 0,
}
`;

exports[`Search Actions 12. verify SEARCH_SUCCESS when mode is 2 1`] = `
Object {
  "all": Array [],
  "cash": Array [],
  "commodity": Array [],
  "currency": Array [],
  "enableLog": false,
  "event": undefined,
  "fno": Array [],
  "index": 3,
  "loading": false,
  "news": Array [],
  "recentSearchVisible": false,
  "searchText": "abs",
  "selectedTab": 0,
}
`;

exports[`Search Actions 13. verify SEARCH_SUCCESS when mode is 3 1`] = `
Object {
  "all": Array [],
  "cash": Array [],
  "commodity": Array [],
  "currency": Array [],
  "enableLog": false,
  "event": undefined,
  "fno": Array [],
  "index": 3,
  "loading": false,
  "news": Array [],
  "recentSearchVisible": false,
  "searchText": "abs",
  "selectedTab": 0,
}
`;

exports[`Search Actions 14. verify SEARCH_SUCCESS when mode is 4 1`] = `
Object {
  "all": Array [],
  "cash": Array [],
  "commodity": Array [],
  "currency": Array [],
  "enableLog": false,
  "event": undefined,
  "fno": Array [],
  "index": 3,
  "loading": false,
  "news": Array [],
  "recentSearchVisible": false,
  "searchText": "abs",
  "selectedTab": 0,
}
`;

exports[`Search Actions 15. verify SEARCH_FAILURE when mode is 0 1`] = `
Object {
  "all": Object {
    "result": Array [],
  },
  "cash": Array [],
  "commodity": Array [],
  "currency": Array [],
  "enableLog": false,
  "event": undefined,
  "fno": Array [],
  "index": 3,
  "loading": false,
  "news": Array [],
  "recentSearchVisible": false,
  "searchText": "abs",
  "selectedTab": 0,
}
`;

exports[`Search Actions 16. verify SEARCH_FAILURE when mode is 1 1`] = `
Object {
  "all": Array [],
  "cash": Object {
    "result": Array [
      Array [
        1,
        2,
      ],
    ],
  },
  "commodity": Array [],
  "currency": Array [],
  "enableLog": false,
  "event": undefined,
  "fno": Array [],
  "index": 3,
  "loading": false,
  "news": Array [],
  "recentSearchVisible": false,
  "searchText": "abs",
  "selectedTab": 0,
}
`;

exports[`Search Actions 17. verify SEARCH_FAILURE when mode is 2 1`] = `
Object {
  "all": Array [],
  "cash": Array [],
  "commodity": Array [],
  "currency": Array [],
  "enableLog": false,
  "event": undefined,
  "fno": Object {
    "result": Array [],
  },
  "index": 3,
  "loading": false,
  "news": Array [],
  "recentSearchVisible": false,
  "searchText": "abs",
  "selectedTab": 0,
}
`;

exports[`Search Actions 18. verify SEARCH_FAILURE when mode is 3 1`] = `
Object {
  "all": Array [],
  "cash": Array [],
  "commodity": Array [],
  "currency": Object {
    "result": Array [],
  },
  "enableLog": false,
  "event": undefined,
  "fno": Array [],
  "index": 3,
  "loading": false,
  "news": Array [],
  "recentSearchVisible": false,
  "searchText": "abs",
  "selectedTab": 0,
}
`;

exports[`Search Actions 19. verify SEARCH_FAILURE when mode is 4 1`] = `
Object {
  "all": Array [],
  "cash": Array [],
  "commodity": Object {
    "result": Array [],
  },
  "currency": Array [],
  "enableLog": false,
  "event": undefined,
  "fno": Array [],
  "index": 3,
  "loading": false,
  "news": Array [],
  "recentSearchVisible": false,
  "searchText": "abs",
  "selectedTab": 0,
}
`;

exports[`Search Actions 20. verify CLEAR_SEARCH_STATE 1`] = `
Object {
  "all": Array [],
  "cash": Array [],
  "commodity": Array [],
  "currency": Array [],
  "fno": Array [],
  "index": 3,
  "loading": Object {
    "key": 0,
    "state": false,
  },
  "news": Array [],
  "recentSearchVisible": true,
  "searchText": "",
  "selectedTab": 0,
}
`;

exports[`Search Actions 21. verify NEWS_SUCCESS 1`] = `
Object {
  "all": Array [],
  "cash": Array [],
  "commodity": Array [],
  "currency": Array [],
  "enableLog": false,
  "event": Object {},
  "fno": Array [],
  "index": 1,
  "loading": false,
  "news": Array [],
  "recentSearchVisible": true,
  "searchText": "abs",
  "selectedTab": 0,
}
`;

exports[`Search Actions 22. verify default case 1`] = `
Object {
  "all": Array [],
  "cash": Array [],
  "commodity": Array [],
  "currency": Array [],
  "enableLog": false,
  "event": Object {},
  "fno": Array [],
  "index": 3,
  "loading": false,
  "news": Array [],
  "recentSearchVisible": true,
  "searchText": "abs",
  "selectedTab": 0,
}
`;

exports[`Search Actions 23. verify CLEAR_RESULT 1`] = `
Object {
  "all": Array [],
  "cash": Array [],
  "commodity": Array [],
  "currency": Array [],
  "enableLog": false,
  "event": Object {},
  "fno": Array [],
  "index": 3,
  "loading": false,
  "news": Array [],
  "recentSearchVisible": true,
  "searchText": "abs",
  "selectedTab": 0,
}
`;

exports[`Search Actions 24. verify SET_SEARCH_TEXT 1`] = `
Object {
  "all": "Enter min 2 characters",
  "cash": "Enter min 2 characters",
  "commodity": "Enter min 2 characters",
  "currency": "Enter min 2 characters",
  "enableLog": false,
  "event": Object {},
  "fno": "Enter min 2 characters",
  "index": 3,
  "loading": Object {
    "key": undefined,
    "state": false,
  },
  "news": Array [],
  "recentSearchVisible": false,
  "searchText": "",
  "selectedTab": 0,
}
`;

exports[`Search Actions 25. verify search 1`] = `
Object {
  "all": Array [],
  "cash": Array [],
  "commodity": Array [],
  "currency": Array [],
  "enableLog": false,
  "event": Object {},
  "fno": Array [],
  "index": 3,
  "loading": Object {
    "key": 0,
    "state": true,
  },
  "news": Array [],
  "recentSearchVisible": false,
  "searchText": "as",
  "selectedTab": 1,
}
`;

exports[`Search Actions 26. verify search 1`] = `
Object {
  "all": Array [],
  "cash": Array [],
  "commodity": Array [],
  "currency": Array [],
  "enableLog": Array [],
  "event": Object {},
  "fno": Array [],
  "index": 3,
  "loading": false,
  "news": Array [],
  "recentSearchVisible": true,
  "searchText": "",
  "selectedTab": 0,
}
`;

exports[`Search Actions 26. verify search 2`] = `
Object {
  "all": Array [],
  "cash": Array [],
  "commodity": Array [],
  "currency": Array [],
  "enableLog": false,
  "event": Object {},
  "fno": Array [],
  "index": 3,
  "loading": false,
  "news": Array [],
  "recentSearchVisible": true,
  "searchText": "",
  "selectedTab": 0,
}
`;
